<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SJF</class>
 <widget class="QWidget" name="SJF">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1317</width>
    <height>656</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>短进程优先算法</string>
  </property>
  <widget class="QLabel" name="finish_list">
   <property name="geometry">
    <rect>
     <x>1070</x>
     <y>130</y>
     <width>72</width>
     <height>15</height>
    </rect>
   </property>
   <property name="text">
    <string>完成队列:</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>890</x>
     <y>630</y>
     <width>72</width>
     <height>15</height>
    </rect>
   </property>
   <property name="text">
    <string>系统时间:</string>
   </property>
  </widget>
  <widget class="QTableWidget" name="run_table">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>400</y>
     <width>471</width>
     <height>56</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="focusPolicy">
    <enum>Qt::StrongFocus</enum>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::DefaultContextMenu</enum>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Sunken</enum>
   </property>
   <property name="rowCount">
    <number>0</number>
   </property>
   <property name="columnCount">
    <number>0</number>
   </property>
   <attribute name="horizontalHeaderDefaultSectionSize">
    <number>117</number>
   </attribute>
  </widget>
  <widget class="QTableWidget" name="end_table">
   <property name="geometry">
    <rect>
     <x>940</x>
     <y>410</y>
     <width>351</width>
     <height>211</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
  </widget>
  <widget class="QLabel" name="wait_list">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>490</y>
     <width>61</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>阻塞队列:</string>
   </property>
  </widget>
  <widget class="QTableWidget" name="arm_table">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>410</y>
     <width>281</width>
     <height>211</height>
    </rect>
   </property>
  </widget>
  <widget class="QWidget" name="widget_3" native="true">
   <property name="geometry">
    <rect>
     <x>1040</x>
     <y>20</y>
     <width>131</width>
     <height>101</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QPushButton" name="begin">
      <property name="text">
       <string>开始</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="stop_run">
      <property name="text">
       <string>暂停</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="continue_run">
      <property name="text">
       <string>继续</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QPushButton" name="sleep">
   <property name="geometry">
    <rect>
     <x>550</x>
     <y>410</y>
     <width>89</width>
     <height>24</height>
    </rect>
   </property>
   <property name="text">
    <string>阻塞</string>
   </property>
  </widget>
  <widget class="QLabel" name="creat_list">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>170</y>
     <width>71</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>后备队列:</string>
   </property>
  </widget>
  <widget class="QTableWidget" name="finish_table">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>150</y>
     <width>341</width>
     <height>231</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Sunken</enum>
   </property>
  </widget>
  <widget class="QPushButton" name="wakeup">
   <property name="geometry">
    <rect>
     <x>550</x>
     <y>530</y>
     <width>89</width>
     <height>24</height>
    </rect>
   </property>
   <property name="text">
    <string>唤醒</string>
   </property>
  </widget>
  <widget class="QTableWidget" name="wait_table">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>470</y>
     <width>471</width>
     <height>171</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Sunken</enum>
   </property>
  </widget>
  <widget class="QLabel" name="ready_list">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>170</y>
     <width>68</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>就绪队列:</string>
   </property>
  </widget>
  <widget class="QLabel" name="os_time">
   <property name="geometry">
    <rect>
     <x>970</x>
     <y>630</y>
     <width>72</width>
     <height>15</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QTableWidget" name="ready_table">
   <property name="geometry">
    <rect>
     <x>540</x>
     <y>150</y>
     <width>401</width>
     <height>231</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
  </widget>
  <widget class="QWidget" name="widget_2" native="true">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1037</width>
     <height>140</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QWidget" name="widget_8" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="ram">
         <property name="text">
          <string>内存:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="ram_edit"/>
       </item>
       <item>
        <widget class="QLabel" name="print">
         <property name="text">
          <string>打印机:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="print_edit"/>
       </item>
       <item>
        <widget class="QPushButton" name="set_os">
         <property name="text">
          <string>系统设置</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="pri_change">
         <property name="text">
          <string>是否启用优先级:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string/>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QRadioButton" name="pri_ture">
            <property name="text">
             <string>是</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="pri_false">
            <property name="text">
             <string>否</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widget_9" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_9"/>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QWidget" name="widget" native="true">
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QLabel" name="name">
         <property name="text">
          <string>进程名:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="name_edit"/>
       </item>
       <item>
        <widget class="QLabel" name="arr">
         <property name="text">
          <string>到达时间:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="arr_edit"/>
       </item>
       <item>
        <widget class="QLabel" name="ser">
         <property name="text">
          <string>服务时间:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="ser_edit"/>
       </item>
       <item>
        <widget class="QLabel" name="arm">
         <property name="text">
          <string>内存:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="arm_edit"/>
       </item>
       <item>
        <widget class="QLabel" name="printer">
         <property name="text">
          <string>打印机:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="printer_edit"/>
       </item>
       <item>
        <widget class="QLabel" name="pri">
         <property name="text">
          <string>优先级:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="pri_edit"/>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="add">
         <property name="text">
          <string>添加进程</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QTableWidget" name="creat_table">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>150</y>
     <width>391</width>
     <height>231</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>660</x>
     <y>390</y>
     <width>72</width>
     <height>15</height>
    </rect>
   </property>
   <property name="text">
    <string>内存：</string>
   </property>
  </widget>
  <widget class="QLabel" name="run_list">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>400</y>
     <width>61</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>执行进程:</string>
   </property>
  </widget>
  <widget class="QLabel" name="have_printer">
   <property name="geometry">
    <rect>
     <x>660</x>
     <y>630</y>
     <width>81</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>剩余打印机:</string>
   </property>
  </widget>
  <widget class="QLabel" name="ending">
   <property name="geometry">
    <rect>
     <x>940</x>
     <y>390</y>
     <width>72</width>
     <height>15</height>
    </rect>
   </property>
   <property name="text">
    <string>运行结果:</string>
   </property>
  </widget>
  <widget class="QPushButton" name="back">
   <property name="geometry">
    <rect>
     <x>1190</x>
     <y>60</y>
     <width>89</width>
     <height>24</height>
    </rect>
   </property>
   <property name="text">
    <string>返回</string>
   </property>
  </widget>
  <widget class="QLabel" name="printer_num">
   <property name="geometry">
    <rect>
     <x>730</x>
     <y>630</y>
     <width>72</width>
     <height>15</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="reback">
   <property name="geometry">
    <rect>
     <x>1190</x>
     <y>30</y>
     <width>89</width>
     <height>24</height>
    </rect>
   </property>
   <property name="text">
    <string>重置</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
